### ISSUE TITLE
crash when `cmdheight=0` and `:intro` is executed

### ISSUE THREAD

**Issue by omar25h** [neovim/neovim#26505]
### Problem

Neovim crashes when `cmdheight=0` and the `:intro` command is executed. This does not happen when `cmdheight` is set to any positive number

```
Assertion failed: (to_scroll >= 0), function msg_scroll_flush, file message.c, line 2345.
```

### Steps to reproduce

nvim --clean
:set cmdheight=0
:intro

### Expected behavior

Intro is shown and nvim does not crash

### Neovim version (nvim -v)

NVIM v0.10.0-dev-1801+g2ebd328a7

### Vim (not Nvim) behaves the same?

no

### Operating system/version

macOS 14.1.2

### Terminal name/version

Wezterm

### $TERM environment variable

wezterm

### Installation

bob

**Comment by clason** [comment]
Does `bob` install a Debug build? I can't reproduce the crash on Release.

**Comment by glepnir** [comment]
I can reproduce. not sure this is correct.

```diff
-    assert(to_scroll >= 0);
+    if (p_ch > 0) {
+      assert(to_scroll >= 0);
+    }
```

**Comment by zeertzjq** [comment]
Bisected to 3a7cb72dc

**Comment by zeertzjq** [comment]
This seems to fix the problem:
```diff
diff --git a/src/nvim/version.c b/src/nvim/version.c
index 2caf2c0cb..470e0b421 100644
--- a/src/nvim/version.c
+++ b/src/nvim/version.c
@@ -2831,6 +2831,11 @@ void intro_message(int colon)
       }
     }
   }
+
+  // Make the wait-return message appear just below the text.
+  if (colon) {
+    msg_row = row;
+  }
 }
 
 static void do_intro_line(int row, char *mesg, int attr)
```

**Comment by zeertzjq** [comment]
Fixed by #28343
